### PR TITLE
add xgboost test for binary classification case with TreeExplainer

### PR DESCRIPTION
**Pull request by imatiach-msft**
validates issue to be fixed in https://github.com/slundberg/shap/pull/1046

### CODE REVIEW

File: test/test_explain_model.py [hunk 1]
```diff
@@ -346,21 +346,32 @@ def test_explain_model_xgboost_multiclass(self, tabular_explainer, iris):
...
         explanation = exp.explain_global(x_test)
         assert len(explanation.local_importance_values[0]) == len(x_test)
         assert len(explanation.local_importance_values) == len(classes)
 
+    @pytest.mark.skip
+    def test_explain_model_xgboost_binary(self, tabular_explainer):
+        # Fit an xgboost tree model
+        def create_model(x_train, y_train):
+            return create_xgboost_classifier(x_train, y_train)
+        self._validate_binary_explain_model(tabular_explainer, create_model, 'test_explain_model_xgboost_binary')
```
Comment (wamartin-aml) [inline test/test_explain_model.py]: I don't like manually passing in a string for the name of the test. Might not be any alternatives though.

File: test/test_explain_model.py [hunk 2]
```diff
@@ -346,21 +346,32 @@ def test_explain_model_xgboost_multiclass(self, tabular_explainer, iris):
...
         # Create tabular explainer
         exp = tabular_explainer(model, x_train, features=X.columns.values,
                                 classes=classes)
-        test_logger.info('Running explain global for test_explain_model_lightgbm_binary')
+        test_logger.info('Running explain global for {}'.format(test_name))
         explanation = exp.explain_global(x_test)
         assert len(explanation.local_importance_values[0]) == len(x_test)
         assert len(explanation.local_importance_values) == len(classes)
 
+    @pytest.mark.skip
```
Comment (wamartin-aml) [inline test/test_explain_model.py]: Why add this test if it's going to be skipped from the start?

File: test/test_explain_model.py (hunk 1 above)
Comment (imatiach-msft) [reply test/test_explain_model.py]: yeah this is just for logging.  Maybe there is a better way.

File: test/test_explain_model.py (hunk 2 above)
Comment (imatiach-msft) [reply test/test_explain_model.py]: Once the fix is released in shap then we can re-enable the test.  But at least for now it's clear that there is an issue.

**Pull request merged**